File: lesson_8_unix/learning-c/playground.js
let obj = { name: "Joe" };
let num = 700;

function modify(o, n) {
  o.name = "Dylan";
  n = 1000;
}

console.log(obj);
console.log(num);
modify(obj, num);
console.log(obj);
console.log(num);

File: lesson_8_unix/learning-c/playground.c
#include <stdio.h>

struct Object {
    const char *name;
};

void modify(struct Object * o, int n) {
    (*o).name = "Dylan";
    n = 1000;
}

int main() {
    struct Object obj;
    obj.name = "Joe";

    int num = 700;

    printf("Before modification:\n");
    printf("obj.name: %s\n", obj.name);
    printf("num: %d\n", num);

    modify(&obj, num);

    printf("\nAfter modification:\n");
    printf("obj.name: %s\n", obj.name);
    printf("num: %d\n", num);

    return 0;
}